From f3fb3a9f9b387c47ed9180830d3bd790d1d70f9a Mon Sep 17 00:00:00 2001
From: "Radivoje (Ogi) Jovanovic" <rjovanovic@nvidia.com>
Date: Fri, 22 Mar 2024 18:55:39 -0700
Subject: [PATCH] [meta-gh] TMP avoid some HTTP 500 responses in Redfish

Avoid observed HTTP 500 responses soon after power transitions
in /redfish/v1/Systems/<system>.  Need a real fix later.

Tested:   Build and ran reboot testing on Starship

Fixes NVBUG https://nvbugs/4084486 (TEMPORARILY)

Signed-off-by: Radivoje (Ogi) Jovanovic <rjovanovic@nvidia.com>
---
 redfish-core/include/utils/sw_utils.hpp | 4 +++-
 redfish-core/lib/systems.hpp            | 6 ++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/redfish-core/include/utils/sw_utils.hpp b/redfish-core/include/utils/sw_utils.hpp
index 95e690c4..822b59fa 100644
--- a/redfish-core/include/utils/sw_utils.hpp
+++ b/redfish-core/include/utils/sw_utils.hpp
@@ -153,7 +153,9 @@ inline void populateSoftwareInformation(
                         {
                             return;
                         }
-                        messages::internalError(asyncResp->res);
+                        // Don't return HTTP 500 here
+                        //messages::internalError(aResp->res);
+
                         return;
                     }
                     // example propertiesList
diff --git a/redfish-core/lib/systems.hpp b/redfish-core/lib/systems.hpp
index a5f81a7c..c56aafd0 100644
--- a/redfish-core/lib/systems.hpp
+++ b/redfish-core/lib/systems.hpp
@@ -352,7 +352,8 @@ inline void
         if (ec2)
         {
             BMCWEB_LOG_ERROR("DBUS response error {}", ec2);
-            messages::internalError(asyncResp->res);
+            //Don't return HTTP 500 here
+            //messages::internalError(asyncResp->res);
             return;
         }
         processMemoryProperties(asyncResp, service, path, properties);
@@ -467,7 +468,8 @@ inline void
                             if (ec3)
                             {
                                 BMCWEB_LOG_ERROR("DBUS response error {}", ec3);
-                                messages::internalError(asyncResp->res);
+                                //Don't return HTTP 500 here
+                                //messages::internalError(asyncResp->res);
                                 return;
                             }
                             BMCWEB_LOG_DEBUG("Got {} UUID properties.", properties.size());
-- 
2.43.0

